samples/scanner/scanners/firewall_rules/matches.yaml

```diff
@@ -0,0 +1,114 @@
+# This file contains several example blacklist firewall rules.
+# 
+# Note: Required and Matches rules can only have "match_policies" fields which will
+# be validated.
+#
+# Matches rules are evaluated in the following steps:
+# 1. Get list of all pairs of resources and policies that apply to the resource.
+# 2. For each pair, get a list of applicable rules by:
+#   a. Checking if a rule applies to the specific resource.
+#   b. Checking each parent in the hierarchy for a rule that applies to that parent.
+#     See: https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy
+#   c. Return the first applicable rules.
+# 3. Check if the policy matches all of the "match_policies" defined in the rule.
+#     If it does not, then this creates a RuleViolation.
+rules:
+
+    # This rule will trigger on policies that do not match the required policy.
+    #
+    # Example triggering firewall rules:
+    # [
+    #   {
+    #     "name": "firewall1",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": [
+    #       "10.0.0.0/8",
+    #       ],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "tcp",
+    #         "ports": ["22"]
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   },
+    #   {
+    #     "name": "firewall2",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": [
+    #       "10.0.0.0/8",
+    #       ],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "tcp",
+    #         "ports": ["80", "443"]
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   },
+    # ]
+    # Example firewall rule that WON'T trigger:
+    # [
+    #   {
+    #     "name": "firewall1",
```

Comment:
This example seems to be the same as the above example that will trigger.  Is the targetTags needed here?

Reply:
The triggering rule doesn't have the targetTags, which is what causes it to fail. The second example has them so the policies match what the rule expects and therefore won't trigger.